File: structural/proxy-example.py
import mysql.connector


class DatabaseConnection:
    def __init__(self):
        self.__mydb = mysql.connector.connect(
            host="localhost",
            user="your_username",
            password="your_password"
        )

    def run_select_query(self, query):
        cursor = self.__mydb.cursor()
        cursor.execute(query)
        result = cursor.fetchall()
        return result


if __name__ == "__main__":
    db = DatabaseConnection()
    db.run_select_query("SELECT * FROM customers")

File: structural/proxy-example.js
const mysql = require('mysql2');

class DatabaseConnection {
    static #myConnector;

    constructor() {
        DatabaseConnection.#myConnector = mysql.createConnection({
            host: "localhost",
            user: "yourusername",
            password: "yourpassword"
        })
    }

    runSelectQuery(query) {
        DatabaseConnection.#myConnector.connect(function (err) {
            if (err) throw err;
            DatabaseConnection.#myConnector.query("SELECT * FROM customers", function (err, result, fields) {
                if (err) throw err;
                console.log(result);
                return result;
            });
        });
    }
}

const db = new DatabaseConnection()
db.runSelectQuery("SELECT * FROM customers")